### DISCUSSION TITLE
speed up lstat for things like "git status"?

### DISCUSSION THREAD

**Discussion by flying-x** [juicedata/juicefs#4853] [Ideas]
First of all I just started using juicefs in GCP. So far it works great and I really appreciate your work. I feel this piece of software is well architected and well made.

It seems though commands like git status is slow with juicefs due to a lot of lstat() calls to the metadata server. Here is an idea:

* how about have a bit in the metadata server for the version of of the last change? If the client detected that it has the latest version of the FS metadata, it can skip getting all the info per inode? This could significantly speed up lstat() in the common case when nothing has changed?

Also, on GCP, I noticed juicefs sync is behavior strangely. When I sync a file like:

juicefs sync gs://bucket_name/dir_name /jfs/dir_name

2 strange things feels like bugs:

1. syncing speed is super slow, like 100Kb/s
2. the dest dir end up being /jfs/dir_name-16

I don't know where does the suffix "-16" came from really.

**Comment by zhoucheng361** [comment]
if you sync directory, you should add a slash after the directory name, such as juicefs sync gs://bucket_name/dir_name/ /jfs/dir_name/,  if without slash, it will match all the file and dir names started with dir_name.

**Comment by zhoucheng361** [comment]
`how about have a bit in the metadata server for the version of of the last change? If the client detected that it has the latest version of the FS metadata, it can skip getting all the info per inode? This could significantly speed up lstat() in the common case when nothing has changed? `

This is what meta cache do.

**Comment by flying-x** [comment]
Where do I find more information on meta cache?
